modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -256,21 +264,23 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
 
         internals.clear();
 
-        int inpCn = inputs[0][1];
-        int inpH = inputs[0][2];
-        int inpW = inputs[0][3];
+        CV_Assert(inputs.size() != 0);
+        std::vector<int> inpShape;
+        for (int i = 2; i < inputs[0].size(); i++) {
+            inpShape.push_back(inputs[0][i]);
+        }
 
+        std::vector<int> outShape;
         int outCn = blobs[0].size[0];
-        Size out;
-
+        int inpCn = inputs[0][1];
         if (padMode.empty())
         {
-            out.height = (inpH + 2 * pad.height - (dilation.height * (kernel.height - 1) + 1)) / stride.height + 1;
-            out.width = (inpW + 2 * pad.width - (dilation.width * (kernel.width - 1) + 1)) / stride.width + 1;
+            for (int i = 0; i < inpShape.size(); i++)
+                outShape.push_back((inpShape[i] + pads[i] + pads[i + pads.size() / 2] - dilations[i] * (kernel_size[i] - 1) - 1) / strides[i] + 1);
```

Comment:
I think it also make sense for as to split paddings vector to begins and ends as IE expects. Can you try?